### PR TITLE
Make measurements compile for ESP32

### PR DESCRIPTION
**Pull request by sirkrypt0**
Depends on #28

### CODE REVIEW

**Review by frcroth — COMMENTED** [review]
Oops wrong PR. Can't delete this.

**Comment by felix-gohla** [comment]
This PR contains a WIP commit. Shall this be reviewed?

**Comment by felix-gohla** [comment]
Should this really be merged into `example/nrf`?

**Comment by felix-gohla** [comment]
Please also implement nano second conversion (see esp_rom_get_cpu_ticks_per_us)

File: examples/esp32/main/clock_cycles.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    clock_cycle_start = esp_cpu_get_cycle_count();
+}
+
+/**
+ * @brief Stops counting clock cycles and returns the number of elapsed cycles.
+ * 
+ */
+static inline uint64_t clock_stop_counting() {
+    esp_cpu_cycle_count_t end = esp_cpu_get_cycle_count();
+    uint64_t val = end - start;
```
Comment (felix-gohla) [inline examples/esp32/main/clock_cycles.h:30]:
```suggestion
    uint64_t val = end - clock_cycle_start;
```

**Review by felix-gohla — CHANGES_REQUESTED** [review]

File: examples/esp32/components/hw_crypto/hw_crypto.c [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+#ifdef CONFIG_USE_HW_CRYPTO
+#include "fido.h"
+
+#include <stdio.h>
+#include <mbedtls/aes.h>
+#include <mbedtls/gcm.h>
+#include <mbedtls/sha256.h>
+#include <mbedtls/sha512.h>
+
+int sha256(const uint8_t *data, size_t data_len, uint8_t *hash) {
```
Comment (felix-gohla) [inline examples/esp32/components/hw_crypto/hw_crypto.c:20]: Those could be static, couldn't they?

File: examples/measurements/common/gpio.c [hunk 3]
```diff
@@ -56,6 +56,73 @@ void delay(double ms) {
     k_sleep(K_MSEC(ms));
 }
 
+#elif defined(ESP_PLATFORM)
```
Comment (felix-gohla) [inline examples/measurements/common/gpio.c:59]: Btw: Could you please add a `#pragma once` to the `gpio.h` file?

**Review by felix-gohla — CHANGES_REQUESTED** [review]
Thanks! :) I think, it doesn't compile for AVR anymore?

File: examples/esp32/components/hw_crypto/hw_crypto.c (hunk 2 above)
Comment (sirkrypt0) [reply examples/esp32/components/hw_crypto/hw_crypto.c:20]: As they are only referenced in the init function below, yes I guess. Adjusted that for the nRF as well.

**Review by felix-gohla — APPROVED** [review]

**Pull request merged**